Fuzzy filter on export bus crashes when filter item is indestructable

**Describe the bug**

When having a an item indestructable (from other mod) that is normally destructable as a filter for the export bus, and having a fuzzy card installed, an exception that crashes the server is raised when used.

Since the item is not exported from the system after crash, the server crashes immediately upon next start.

Running 8.3.0-beta.1

**To Reproduce**

1. Install mods:
-- Applied Energetics 2 - version 8.3.0-beta.1
-- Apotheosis (version 4.5.1 for me, probably other mods providing the same functionality might work)
-- Mekanism Generators (only for infinite power source)
2. Run in creative
3. Create a minimal ME system (infite power source + ME controller + ME interface + disk drive)
4. Create a normal chest, connect an export bus + fuzzy card
5. Create an indestructable pickaxe using `/affixloot ANCIENT PICKAXE`. Will create an unbreakable diamond pickaxe from the "Apotheosis" mod.
6. Add the indestructable pickaxe to the export bus as filter
7. Create a normal pickaxe and put in the ME system

**What I see**
- System crashes with exception:

```
Caused by: java.lang.IllegalStateException: ItemStack#isDamageable() has to be true
	at com.google.common.base.Preconditions.checkState(Preconditions.java:444) ~[guava-21.0.jar:?]
	at appeng.util.item.FuzzyItemVariantList.makeLowerBound(FuzzyItemVariantList.java:148) ~[?:8.3.0-beta.1]
	at appeng.util.item.FuzzyItemVariantList.findFuzzy(FuzzyItemVariantList.java:51) ~[?:8.3.0-beta.1]
	at appeng.util.item.ItemList.findFuzzy(ItemList.java:64) ~[?:8.3.0-beta.1]
	at appeng.util.item.ItemList.findFuzzy(ItemList.java:37) ~[?:8.3.0-beta.1]
	at appeng.parts.automation.ExportBusPart.doBusWork(ExportBusPart.java:155) ~[?:8.3.0-beta.1]
	at appeng.parts.automation.ExportBusPart.tickingRequest(ExportBusPart.java:216) ~[?:8.3.0-beta.1]
	at appeng.me.cache.TickManagerCache.onUpdateTick(TickManagerCache.java:84) ~[?:8.3.0-beta.1]
	... 17 more
``` 

<!-- Please provide the steps to reproduce it, otherwise we might not be able to fix it. -->

**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->

Except the new pickaxe to be exported to the chest

**Additional context**

Log: https://pastebin.com/eddxiE8z

**Environment**

Crash report: https://pastebin.com/W90nRBWE

**Notes**

Probably caused by 0bdc408e950fe787277d1e48be61ebf928127429 when the `makeLowerBound()` function added a precondition rather than defaulting the damage value in case `isDamageable()` is false.

--- a/src/main/java/appeng/util/item/FuzzyItemVariantList.java
+++ b/src/main/java/appeng/util/item/FuzzyItemVariantList.java
@@ -145,7 +145,7 @@ class FuzzyItemVariantList extends ItemVariantList {
      * higher number than the upper bound.
      */
     static ItemDamageBound makeLowerBound(final ItemStack stack, final FuzzyMode fuzzy) {
-        Preconditions.checkState(stack.isDamageable(), "ItemStack#isDamageable() has to be true");
+        Preconditions.checkState(stack.getItem().isDamageable(), "Item#isDamageable() has to be true");
 
         int damage;
         if (fuzzy == FuzzyMode.IGNORE_ALL) {
@@ -163,7 +163,7 @@ class FuzzyItemVariantList extends ItemVariantList {
      * lower number than the lower bound. It also is exclusive.
      */
     static ItemDamageBound makeUpperBound(final ItemStack stack, final FuzzyMode fuzzy) {
-        Preconditions.checkState(stack.isDamageable(), "ItemStack#isDamageable() has to be true");
+        Preconditions.checkState(stack.getItem().isDamageable(), "Item#isDamageable() has to be true");
 
         int damage;
         if (fuzzy == FuzzyMode.IGNORE_ALL) {

--- a/src/test/java/appeng/util/item/FuzzyItemVariantListTest.java
+++ b/src/test/java/appeng/util/item/FuzzyItemVariantListTest.java
@@ -2,6 +2,7 @@ package appeng.util.item;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
 
 import java.util.Arrays;
 
@@ -21,6 +22,13 @@ public class FuzzyItemVariantListTest {
         ItemStack undamagedSword = new ItemStack(Items.DIAMOND_SWORD);
         AESharedItemStack undamagedStack = new AESharedItemStack(undamagedSword);
 
+        // Unbreakable Diamond Sword @ 50% durability
+        ItemStack unbreakableSword = new ItemStack(Items.DIAMOND_SWORD);
+        unbreakableSword.setDamage(unbreakableSword.getMaxDamage() / 2);
+        unbreakableSword.getOrCreateTag().putBoolean("Unbreakable", true);
+        assertFalse(unbreakableSword.isDamageable());
+        AESharedItemStack unbreakableStack = new AESharedItemStack(unbreakableSword);
+
         // Unenchanted Diamond Sword @ 0% durability
         ItemStack damagedSword = new ItemStack(Items.DIAMOND_SWORD);
         damagedSword.setDamage(damagedSword.getMaxDamage());
@@ -28,10 +36,10 @@ public class FuzzyItemVariantListTest {
 
         // Create a list of stacks and sort by their natural order
         AESharedItemStack[] stacks = new AESharedItemStack[] {
-                damagedStack, undamagedStack
+                damagedStack, undamagedStack, unbreakableStack
         };
         Arrays.sort(stacks, FuzzyItemVariantList.COMPARATOR);
-        assertThat(stacks).containsExactly(damagedStack, undamagedStack);
+        assertThat(stacks).containsExactly(damagedStack, unbreakableStack, undamagedStack);
     }
 
     @Nested
@@ -49,6 +57,20 @@ public class FuzzyItemVariantListTest {
             DamageBounds bounds = new DamageBounds(stack, FuzzyMode.IGNORE_ALL);
             assertEquals(stack.getMaxDamage(), bounds.lower.itemDamage);
             assertEquals(-1, bounds.upper.itemDamage);
+        }
+
+        /**
+         * Unbreakable items are still considered at their current damage value.
+         */
+        @Test
+        void test99PercentDurabilityWithUnbreakable() {
+            ItemStack unbreakableStack = damagedStack.copy();
+            unbreakableStack.getOrCreateTag().putBoolean("Unbreakable", true);
+            assertFalse(unbreakableStack.isDamageable());
+
+            DamageBounds bounds = new DamageBounds(unbreakableStack, FuzzyMode.PERCENT_99);
+            assertEquals(stack.getMaxDamage(), bounds.lower.itemDamage);
+            assertEquals(0, bounds.upper.itemDamage);
         }
 
         /**

